File: cli/commands/list_models.py
# CLI command to list models for the current provider
from core.config import settings
from providers.factory import get_provider
from utils import Colors

def main():
	provider_name = settings.get('PROVIDER', 'openai')
	print(Colors.header(f"🤖 Fetching available models for {provider_name}..."))
	
	try:
		provider = get_provider(provider_name)
		models = provider.list_models()
		
		print(Colors.success(f"📋 Available models for {provider_name}:"))
		for i, model in enumerate(models, 1):
			print(Colors.highlight(f"  {i}. {model}"))
		
		if not models:
			print(Colors.warning("⚠️ No models found or model list is empty"))
		else:
			print(Colors.dim(f"\n💡 Total: {len(models)} model(s) available"))
			print(Colors.dim("💡 Use these model names in your configuration"))
			
	except Exception as e:
		print(Colors.error(f"❌ Error fetching models: {e}"))
		print(Colors.dim("💡 Check your provider configuration and network connection"))

if __name__ == "__main__":
	main()

File: src/cli/list_models.rs
use crate::config::AppConfig;
use crate::services::ai::AiClient;
use crate::utils::colors::Colors;
use crate::utils::errors::AppResult;

pub fn run(config: AppConfig) -> AppResult<()> {
    let provider = config.provider();
    config.ensure_provider_ready(&provider)?;
    let ai = AiClient::new(config)?;

    println!(
        "{}",
        Colors::header(format!("[net] fetching models for {provider}"))
    );

    match ai.list_models(&provider) {
        Ok(models) => {
            println!(
                "{}",
                Colors::success(format!("[models] available models for {provider}"))
            );
            for (idx, model) in models.iter().enumerate() {
                println!("{}", Colors::highlight(format!("  {}. {}", idx + 1, model)));
            }
            if models.is_empty() {
                println!(
                    "{}",
                    Colors::warning("[warn] no models found or model list is empty")
                );
            } else {
                println!(
                    "{}",
                    Colors::dim(format!("\n[info] {} model(s) available", models.len()))
                );
                println!(
                    "{}",
                    Colors::dim("[hint] use these model names in your configuration")
                );
            }
        }
        Err(err) => {
            println!(
                "{}",
                Colors::error(format!(
                    "[err] failed to fetch models: {}",
                    err.user_message()
                ))
            );
            println!(
                "{}",
                Colors::dim("[hint] check provider configuration and network access")
            );
        }
    }

    Ok(())
}
